evennia/utils/utils.py

```diff
@@ -2202,18 +2204,24 @@ def attributes(self):
 
     """
 
-    def __init__(self, func, name=None, doc=None):
+    def __init__(self, func: Callable[..., TProp], name=None, doc=None):
         """Store all properties for now"""
         self.__name__ = name or func.__name__
         self.__module__ = func.__module__
         self.__doc__ = doc or func.__doc__
         self.func = func
 
-    def __get__(self, obj, type=None):
+    @overload
+    def __get__(self, obj: None, type=None) -> 'lazy_property': ...
+
+    @overload
+    def __get__(self, obj, type=None) -> TProp: ...
+
+    def __get__(self, obj, type=None) -> TProp | 'lazy_property':
         """Triggers initialization"""
         if obj is None:
             return self
-        value = obj.__dict__.get(self.__name__, _missing)
+        value: TProp = obj.__dict__.get(self.__name__, _missing)
         if value is _missing:
             value = self.func(obj)
         obj.__dict__[self.__name__] = value
```

Comment:
I didn't touch it in case it's this way on purpose, but can this caching be moved into the if block? Seems like a redundant step if it's not missing.